fission/src/ui/panels/configuring/MatchModeConfigPanel.tsx

```diff
@@ -0,0 +1,272 @@
+                display={"flex"}
+                key={`button-box-${id}`}
+                flexDirection={"row-reverse"}
+                gap={"0.25rem"}
+                justifyContent={"center"}
+                alignItems={"center"}
+            >
+                <NegativeButton value={SynthesisIcons.DeleteLarge} onClick={secondaryOnClick} />
+                <PositiveButton value={SynthesisIcons.SelectLarge} onClick={primaryOnClick} />
+            </Box>
+        </Box>
+    )
+}
+
+const MatchModeConfigPanel: React.FC<PanelPropsImpl> = ({ panelId }) => {
+    const { closePanel } = usePanelControlContext()
+    const { openModal } = useModalControlContext()
+
+    const [matchModeConfigs, setMatchModeConfigs] = useState<MatchModeConfig[]>([])
+
+    useEffect(() => {
+        const loadAllJsonFiles = async () => {
+            try {
+                const indexRes = await fetch("match-mode-config/index.json")
+                const fileNames: string[] = await indexRes.json()
+
+                const defaultConfigs = await Promise.all(
+                    fileNames.map(async fileName => {
+                        const res = await fetch(`match-mode-config/${fileName}`)
+                        return res.json()
+                    })
+                )
+                const localConfigs = JSON.parse(window.localStorage.getItem("match-mode-configs") || "[]")
+
+                const combinedConfigs = [...defaultConfigs, ...localConfigs]
+                const uniqueConfigsById = Array.from(new Map(combinedConfigs.map(item => [item.id, item])).values())
+
+                setMatchModeConfigs(uniqueConfigsById)
+            } catch (err) {
+                console.error("Error loading JSON files:", err)
+                Global_AddToast?.(
+                    "error",
+                    "Error Loading Match Mode Configs",
+                    "Please check the console for more details"
+                )
+            }
+        }
+
+        loadAllJsonFiles()
+    }, [])
+
+    const matchModeConfigElements = useMemo(
+        () =>
+            matchModeConfigs.map(config =>
+                ItemCard({
+                    id: config.id,
+                    name: config.name || config.id || "Unnamed Match Mode",
+                    primaryOnClick: () => {
+                        MatchConfigSelected(config, openModal)
+                        closePanel("match-mode-config")
+                    },
+                    secondaryOnClick: () => {
+                        // Delete the config from the local storage
+                        setMatchModeConfigs(prev => prev.filter(c => c.id !== config.id))
+                        window.localStorage.setItem(
+                            "match-mode-configs",
+                            JSON.stringify(matchModeConfigs.filter(c => c.id !== config.id))
+                        )
+                        matchModeConfigs.filter(c => c.id !== config.id)
+                    },
```

Comment:
I think there might be a race condition here since we are reading `matchModeConfigs` when `setMatchModeConfigs` has a chance of not being done updating yet.